spec/server/functional/user.spec.coffee

```diff
@@ -246,7 +246,214 @@ ghlfarghlarghlfarghlarghlfarghlarghlfarghlarghlfarghlarghlfarghlarghlfarghlarghl
     [res, body] = yield request.putAsync { uri: url, json: { role: 'student' }}
     expect(body.role).toBe('advisor')
     done()
-      
+
+describe 'PUT /db/user/-/become-student', ->
+
+  it 'when individual (role unset) in classrooms', utils.wrap (done) ->
+    user = yield utils.initUser()
+    yield utils.loginUser(user)
+    classroom = new Classroom({
+      members: [user._id]
+    })
+    yield classroom.save()
+    url = getURL('/db/user/-/become-student')
+    [res, body] = yield request.putAsync { uri: url}
+    expect(res.statusCode).toEqual(200)
+    user = yield User.findById user.id
+    expect(user.get('role')).toEqual('student')
+    classroom = yield Classroom.findById classroom._id
+    expect(classroom.get('members')[0]).toEqual(user._id)
+    done()
```

Comment:
Tests should be grammatically correct, grouped into contexts (`describe`) and assertions (`it`). This test, for example, might be instead 

```
describe 'PUT /db/user/-/become-student', ->
  describe 'when the user has no role and is in a classroom', ->
    beforeEach utils.wrap (done) ->
      @user = yield utils.initUser()
      yield utils.loginUser(@user)
      @classroom = new Classroom({
        members: [user._id]
      })
      yield @classroom.save()

    it 'keeps the user in their classroom and sets their role to student', utils.wrap (done) ->
      url = getURL('/db/user/-/become-student')
      [res, body] = yield request.putAsync { uri: url}
      expect(res.statusCode).toEqual(200)
      user = yield User.findById user.id
      expect(user.get('role')).toEqual('student')
      classroom = yield Classroom.findById classroom._id
      expect(classroom.get('members')[0]).toEqual(user._id)
      done()
```

This way strings match implementations. `beforeEach`'s code matches its `describe`'s string, and `it`'s code matches the given string. Then the resulting spec makes more sense, if read by itself.

```
PUT /db/user/-/become-student
  when the user has no role and is in a classroom
    it keeps the user in their classroom and sets their role to student
```

Although, if there's just one test in a describe, the 'when' part can just go at the end of the 'it' string and the test can be kept as it was originally.

```
PUT /db/user/-/become-student
  it keeps the user in their classroom and sets their role to student when the user has no role and is in a classroom
```

But then it's getting wordy. It might then just be shortened to 

```
PUT /db/user/-/become-student
  it sets user's role to student when the user has no role
```

And keep out the classroom assertions, leaving the outputted spec more concise.